docs/diagrams/dataplane.drawio

```diff
@@ -0,0 +1 @@
+<mxfile modified="2019-05-03T19:02:32.507Z" host="www.draw.io" agent="Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/9.3.1 Chrome/66.0.3359.181 Electron/3.0.6 Safari/537.36" etag="VrqfSZcEyWRXZgNxOAMM" version="10.6.5" type="device"><diagram id="EbcCn81XiduqCDuKKwkF" name="Page-1">7Vxbb6M4GP01kWYfWnEzkMfpZXZH05FG6ko7fXTAIZ4hGBmnTfrr1+aWYLsJ03JJ0iZSi41N4HxXH2NP7Ovl+m8K08V3EqJ4YhnhemLfTCzLNDyH/xM1m6rGsIqaiOKwrNtW3ONnVDUsa1c4RFmjISMkZjhtVgYkSVDAGnWQUvLUbDYncfNXUxghpeI+gLFa+x8O2aKo9YGxrf8H4WhR/TJ/wOLMElaNy4psAUPytFNl307sa0oIK46W62sUC/QqXIp+X144W98YRQlr0+Fh6dz/vPPvZon5vP7lPP/7/Agu/PIyjzBelU9c3i3bVBBQskpCJK5iTOyrpwVm6D6FgTj7xKXO6xZsGfOSyQ/LyyHK0PrFGzXrx+eKg8gSMbrhTcoOFWCbZvFpC39dt2hCX4q9FHlUX3mLCj8ogfkDkKyjx8gEKki+oQHJ8fsCqWtFCmG2yNuKwhzH8TWJCc0vZIcA+aHD6zNGyW+0c8a3ZrbrdoOxI+mhBmPP1WBsu31h7DodgyzjCpE/D3S4uoGPZvNucHUlXH0NrjrdtXrTXdc+B1w9Cdfp+Lh27RRGwdVv4mobo+NqTXvGdRj/KjlY2wQKsLYu0teeufsgBs5BYYEErKMB1tSmUFZfwKoOFmcMkwucRBRlWQQZeoLiESaWC5cCxOIvr/mdQIYfkbirgB9AxiGzjE94LsQiEvUwHxPwP8+Ikr9O0BDkgGhqHIzWDnrz274irgAmML9nRuF8joNcVOAA2ISyBYlIAuM7QtIS4l+IsU050IMrLriGANAas5+i+yUoSw87Z27W5ZXzwqYqJPyhdzqJ4sPuuW23vFT1U4Rq5B9+JoYzFF/B4HeUP1DVJCEJElcJ+RCxfObtI95ua0ulEO32qwTHjqxogA5bDoM0QmxfTAB7dezCuLSB4zcUrdIziuLCxBqDZ41SlVf/QTB/iq0flwdstqSXxTOWvbaq+ZnS3OSrZqlokP3B73jSgFdq7xp72/OD4g6k3tXtkPk8Q2wi21IN7RuizFSxm8MqNYpldajHoKUem7Zej1sr6Zs8nxr/U4ouUkoCHqUQPcVkAEjJgKVJBrTpa2/Zqz2q+u8o/9YUDgQWsxFWtlHmhcDSodn4bc3mBfc/jNmo+QJapzHEyUmajOsfzp91xE9/6bPpv8OIYbZNfUxnTN031bENjxkhDvKByskpvxIv3LHjhamyninJ2FlFZWCMjbKlokxWLMYCXiNEDJ2oPjvusTlzyx3Fmb9ykNyhQ7dajwFGTWYsb9xg6/XPfIwhVH9UmY6cQFnnKVNr1MzLUkfrhAYLxOMJPI9gZdtqsKqnlRpzzr0R94aC8cRyY/6zVzN+EImD64ob/lZT9dckmeNoxeWASZJVPfgd1J0U6XDUWFMETahL8nVXLmUVjHGU8GLAUebpin0lZIADGH8uTyxxGOYWrZN5Uyu6yO2k4SPQCNHUCbG3jMNxNULs0jgGmSYxJT73COYLHa9nYAfxOpZxOEUeeL7QeY+EhzNtGXbBqBx5dZtnRJLXU2FHQ5JX/MAHSd5mbql6M/iQ4RT+erTJJfV1qFOmye0Wr5kMy6wANfqef9QAbWlyMOpgDZwXTa5EjNFpcnCGNLmM8vg0OdBwDudAk9dO+Xic+XulyUHbYcDI2cwHTd6HUMELdjqQUD948l6EOuqEFlAH7CfOk8vRanye3G3Bk9+gOVzF4h3TD6I8l5BzbES5ax2W4rcv94g+Yo7MuxKWJ6filcY3lkENKCtv/2C/RLGJw6/VMq0aZAtI09fFq5cXDnQYXLy2pNaoZLCnclq3a66vHE5emy/V+ZrU5et4lbF8qHRHYDiDMUyCjgemcyC+2gCUf0QPkrCd+uLTjZVI87empzGSQZfyeLqlrZJH+yqWYvE231G2eGdOzZacmq9GoGGdmm6FtxyA/GzXlN6VvBzQkJerWyo3qLz2kxR9BqHX0M6DBC63beByRx3qeurc/Q1KERdUEmw6DUnDvB4huTJNemYNypRWifsbqITOLaVBLxh7TWX/EtCXDYlwrcCsB7LBe+sLlGVXaZXkVOJ9JXUojFhZu9nVUkhPXTlUjq+SaCJ2dVqmHPDyt0+Lo5Ds0R7bHj2VBdqm4ykljzjMD7MUBbhY4f2JpIKWgHG3q+rHTdEluWh2SRnYT759OWRHfvLoXJ5v9eLypPczTZkU7Nnn+S1IphsMIwqXgipEm/eV45vyGFqT4w9KCvqq4/xRvLnwSfWLJ2a6XRpr20UD3n75i50qhLPXBc/XGn7/ezr46nsBim7UO0AK5Ov97YwWVpuL9wfJsAjHOlO9kxrMCGNkqbFlJhRKNXl+a6m4y+U6Eht5XsLnFUWXqyw/KYVv70Z8OwrGU/cSaEW9Y+3VW9a71g76snZXtfawnrfZv/nN6MOb490yp0M/0/p1t+khN2OBadPNXJz5jjjKTlP974jDi9stbovm252C7dv/AQ==</diagram></mxfile>
```

Comment:
Nice start. I have few things, but best to get it merged first. Let`s get to the MVP

Reply:
Yeah. I plan on building these concepts in soon, this is the first step of a proposal, perhaps I should've started with a doc, but it takes longer to polish.

The high level idea is that there's a static graph of components that come out of the box with KFServing. For the 20% of use cases that require arbitrary graphs or more advanced features, we'll need higher level CRDs (or just use Seldon). We should fit as much as we can (within reason) to the KFServing CRD. I've been thinking about it as follows, "If it's a feature that maps to a predictor on a 1:1 basis (i.e. singleton), it probably makes sense in KFService."


Blackbox explainers often call predictors 1 to n times. Even whitebox explainers often call predictors as well.